src/Redux/books/bookSlice.js

```diff
@@ -0,0 +1,22 @@
+const ADD_BOOK = 'books/bookSlice/ADD_BOOK';
+const DELETE_BOOK = 'books/bookSlice/ADD_BOOK';
+const initialState = [];
+
+export default function bookReducer(state = initialState, action) {
+  switch (action.type) {
+    case ADD_BOOK:
+      return [
+        ...state,
+        action.payload,
+      ];
+    case DELETE_BOOK:
+      return [
+        ...state.filter((item) => item.id !== action.id),
+      ];
+    default:
+      return state;
+  }
+}
+
+export const addBook = (book) => ({ type: ADD_BOOK, payload: book });
```

Comment:
- [ ] Fantastic job on your state management implementation using individual action types and action creators! However, for improved code simplicity and better alignment with project guidelines, I'd suggest adopting Redux Toolkit's createSlice approach. It offers automatic action creator generation, concise syntax, and enhances maintainability. Here is an example implementation.

```
// ./src/redux/books/booksSlice.js

import { createSlice } from "@reduxjs/toolkit"

export const booksSlice = createSlice({
  name: "books",
  initialState: {
    books: [],
  },
  reducers: {
    addBook: (state, action) => {
      state.books = [
        ...state.books,
        {
          id: Math.floor(Math.random() * 100),
          title: action.payload.title,
          author: action.payload.author,
        },
      ]
    },
    removeBook: (state, action) => {
      state.books = state.books.filter((book) => book.id !== action.id)
    },
  },
})

export const { addBook, removeBook } = booksSlice.actions

export default booksSlice.reducer


```